management_api_app/api/routes/workspaces.py

```diff
@@ -20,13 +23,18 @@ async def retrieve_active_workspaces(workspace_repo: WorkspaceRepository = Depen
 
 @router.post("/workspaces", status_code=status.HTTP_202_ACCEPTED, response_model=WorkspaceIdInResponse, name=strings.API_CREATE_WORKSPACE)
 async def create_workspace(workspace_create: WorkspaceInCreate, workspace_repo: WorkspaceRepository = Depends(get_repository(WorkspaceRepository))) -> WorkspaceIdInResponse:
-    workspace = workspace_repo.create_workspace(workspace_create)
+    try:
+        workspace = workspace_repo.create_workspace_item(workspace_create)
+    except ValueError as e:
+        raise HTTPException(status_code=status.HTTP_400_BAD_REQUEST, detail=str(e))
 
     try:
         service_bus = ServiceBus()
         await service_bus.send_resource_request_message(workspace)
-    except Exception:
+        workspace_repo.save_workspace(workspace)
```

Comment:
This is now interesting. You are creating an item and only saving it after sending a service bus message? Any reason why not saving and then sending a message?

Reply:
moving this